interface/main/messages/messages.php

```diff
@@ -1,668 +1,846 @@
-   </select>
-  </td>
-   <td>
-    <input type ="button" name="clear_user" class= "css_button" id = "clear_user" value="<?php echo htmlspecialchars(xl('Clear'), ENT_QUOTES); ?>" >
-</td>
- </tr>
-
-<?php
-if ($noteid) {
-  // Get the related document IDs if any.
-    $tmp = sqlStatement(
-        "SELECT id1 FROM gprelations WHERE " .
-        "type1 = ? AND type2 = ? AND id2 = ?",
-        array('1', '6', $noteid)
-    );
-    if (sqlNumRows($tmp)) {
-        echo " <tr>\n";
-        echo "  <td class='text'><b>";
-        echo xlt('Linked document') . ":</b>\n";
-        while ($gprow = sqlFetchArray($tmp)) {
-            $d = new Document($gprow['id1']);
-            $enc_list = sqlStatement("SELECT fe.encounter,fe.date,openemr_postcalendar_categories.pc_catname FROM form_encounter AS fe ".
-            " left join openemr_postcalendar_categories on fe.pc_catid=openemr_postcalendar_categories.pc_catid  WHERE fe.pid = ? order by fe.date desc", array($prow['pid']));
-            $str_dob = htmlspecialchars(xl("DOB") . ":" . $prow['DOB'] . " " . xl("Age") . ":" . getPatientAge($prow['DOB']));
-            $pname = $prow['fname']." ".$prow['lname'];
-            echo "<a href='javascript:void(0);' ";
-            echo "onClick=\"gotoReport(".addslashes(attr($d->get_id())).",'".addslashes(attr($pname))."',".addslashes(attr($prow['pid'])).",".addslashes(attr($prow['pubpid'])).",'".addslashes(attr($str_dob))."');\">";
-            echo text($d->get_url_file());
-            echo "</a>\n";
+<html>
+<head>
+    <title><?php echo xlt('Message Center'); ?></title>
+
+    <?php Header::setupHeader(['jquery-ui', 'jquery-ui-redmond', 'opener', 'moment']); ?>
+
+    <script>
+        <?php
+        if ($GLOBALS['date_display_format'] == '0') {
+            $date_format = 'yy-m-d';
+        } elseif ($GLOBALS['date_display_format'] == '1') {
+            $date_format = 'mm/dd/yy';
+        } elseif ($GLOBALS['date_display_format'] == '2') {
+            $date_format = 'dd/mm/yy';
         }
-
-        echo "  </td>\n";
-        echo " </tr>\n";
-    }
-
-  // Get the related procedure order IDs if any.
-    $tmp = sqlStatement(
-        "SELECT id1 FROM gprelations WHERE " .
-        "type1 = ? AND type2 = ? AND id2 = ?",
-        array('2', '6', $noteid)
-    );
-    if (sqlNumRows($tmp)) {
-        echo " <tr>\n";
-        echo "  <td class='text'><b>";
-        echo xlt('Linked procedure order') . ":</b>\n";
-        while ($gprow = sqlFetchArray($tmp)) {
-            echo "   <a href='";
-            echo $GLOBALS['webroot'] . "/interface/orders/single_order_results.php?orderid=";
-            echo $gprow['id1'];
-            echo "' target='_blank' onclick='top.restoreSession()'>";
-            echo $gprow['id1'];
-            echo "</a>\n";
+        ?>
+        var xljs_dateFormat = '<?php echo $date_format; ?>';
+        var xljs1 = '<?php echo xl( 'Preferences updated successfully' ); ?>';
+        <?php require_once "$srcdir/restoreSession.php"; ?>
```

Comment:
@ophthal ,
On my testing:
1. I was unable find any places that uses the message screen in a popup.
2. I was unable to find any place in messages screen where a date widget is used.

I just wanted to confirm that I am not missing anything.

thanks,
-brady

Reply:
xljs_dateFormat was used to make the bootstrap datepicker conform to the global for date format.  Since reverting back to original datepicker there is no need to propagate that global value through javascript.

The pop-up part is utilized in the SMS Bot.  That functionality is not visible without enabling MedEx and logging in (use the demo credentials).

Reply:
@ophthal , thanks and very neat. I just sent out a SMS text via that thing. Ok, now I understand that the screen can be used as a popup.(I tested it with a logged in medex user, but I didn't realize that SMS Bot was using messages.php). Gonna now unwind the bootstrap datepicker stuff (I think I have a good idea of where this needs to be done).

Reply:
Yes messages.php is utilized to provide the OAuth authentication layer between the two servers.  I tried to limit the "footprint of intrusion" to messages.php, but wound up affecting flow board too along the way.  It seemed more logical to access an SMS interface from messages.php.